File: backend/src/main/java/com/ai/travelplanner/entity/Expense.java
package com.ai.travelplanner.entity;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "expenses")
public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trip_plan_id", nullable = false)
    private TripPlan tripPlan;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 32)
    private ExpenseCategory category;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false, length = 16)
    private String currency;

    @Column(length = 512)
    private String note;

    @Column(nullable = false)
    private OffsetDateTime recordedAt;
}

File: frontend/src/types/expense.ts
export type ExpenseCategory =
  | 'TRANSPORT'
  | 'HOTEL'
  | 'FOOD'
  | 'ENTERTAINMENT'
  | 'SHOPPING'
  | 'OTHER';

export interface Expense {
  id?: string;
  category: ExpenseCategory;
  amount: number;
  currency: string;
  note?: string;
  recordedAt: string;
  itineraryId?: string;
}

export interface ExpensePayload {
  category: ExpenseCategory;
  amount: number;
  currency: string;
  note?: string;
  recordedAt?: string;
}

